### PR TITLE
query: claimable incentives

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

This PR adds a claimable incentives query. Given a positionID, it returns both the claimable incentives, as well as how many incentives will be forfeited if claimed now.

## Brief Changelog

- Adds QueryClaimableIncentives
- Adds respective tests

## Testing and Verifying

Adds query to both the `TestCollectIncentives` and `TestClaimAllIncentives`.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/client/cli/query.go [hunk 1]
```diff
@@ -43,8 +44,16 @@ func GetCmdPools() (*osmocli.QueryDescriptor, *queryproto.PoolsRequest) {
 
 func GetClaimableFees() (*osmocli.QueryDescriptor, *queryproto.ClaimableFeesRequest) {
 	return &osmocli.QueryDescriptor{
-		Use:   "claimable-fees [poolID] [address] [lowerTick] [upperTick]",
+		Use:   "claimable-fees [positionID]",
 		Short: "Query claimable fees",
 		Long: `{{.Short}}{{.ExampleHeader}}
-{{.CommandPrefix}} claimable-fees 1 osmo12smx2wdlyttvyzvzg54y2vnqwq2qjateuf7thj [-100] 100`}, &queryproto.ClaimableFeesRequest{}
+{{.CommandPrefix}} claimable-fees 53`}, &queryproto.ClaimableFeesRequest{}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/client/cli/query.go:50]: Drive by change

File: x/concentrated-liquidity/incentives_test.go [hunk 2]
```diff
@@ -2908,14 +2908,33 @@ func (s *KeeperTestSuite) TestCollectIncentives() {
...
 
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
+				s.Require().Equal(tc.expectedIncentivesClaimed, incentivesClaimedQuery)
+				s.Require().Equal(tc.expectedForfeitedIncentives, incentivesForfeitedQuery)
+
+				// Ensure balances are unchanged
+				s.Require().Equal(incentivesBalanceBeforeCollect, incentivesBalanceAfterCollect)
+				s.Require().Equal(ownerBalancerAfterCollect, ownerBalancerBeforeCollect)
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives_test.go:2925]: We should probably be checking this even if no error

File: x/concentrated-liquidity/incentives_test.go [hunk 3]
```diff
@@ -3393,14 +3412,32 @@ func (s *KeeperTestSuite) TestClaimAllIncentives() {
...
+			newPoolBalances := s.App.BankKeeper.GetAllBalances(s.Ctx, clPool.GetAddress())
+
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().Error(err)
+				s.Require().ErrorIs(err, tc.expectedError)
+
+				// Ensure balances have not been mutated
+				s.Require().Equal(initSenderBalances, newSenderBalances)
+				s.Require().Equal(initPoolBalances, newPoolBalances)
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives_test.go:3428]: Ditto

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/incentives_test.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/incentives_test.go]: Good point, changed here https://github.com/osmosis-labs/osmosis/pull/5007/commits/6a164cee29e89bf304b495059a8eb4f60add6710

File: x/concentrated-liquidity/incentives_test.go (hunk 3 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/incentives_test.go]: Changed here https://github.com/osmosis-labs/osmosis/pull/5007/commits/6a164cee29e89bf304b495059a8eb4f60add6710

**Review by mattverse — APPROVED** [review]
LGTM

**Pull request merged**